packages/framework-provider-local/src/library/events-adapter.ts

```diff
@@ -65,11 +78,28 @@ export async function storeEvents(
   _config: BoosterConfig,
   logger: Logger
 ): Promise<void> {
-  logger.info('Publishing the following events:', eventEnvelopes)
-
-  for (const event of eventEnvelopes) {
-    await eventRegistry.store(event)
+  logger.debug('[EventsAdapter#storeEvents] Storing the following event envelopes:', eventEnvelopes)
+  for (const eventEnvelope of eventEnvelopes) {
+    await retryIfError(
+      logger,
+      () => persistEvent(eventRegistry, eventEnvelope),
+      OptimisticConcurrencyUnexpectedVersionError
+    )
   }
+  logger.debug('[EventsAdapter#storeEvents] EventEnvelopes stored')
 
   await userApp.boosterEventDispatcher(eventEnvelopes)
 }
+
+async function persistEvent(eventRegistry: EventRegistry, eventEnvelope: EventEnvelope): Promise<void> {
+  try {
+    await eventRegistry.store(eventEnvelope)
+  } catch (e) {
+    //TODO check the exception raised when there is a write error, to implement
+    //Optimistic Concurrency
+    //if (e.name == 'TODO') {
+    //  throw new OptimisticConcurrencyUnexpectedVersionError(e.message)
+    //}
```

Comment:
This will be added at a later stage of the project, since it's not part of the MVP